### PR TITLE
Update Airflow filtered warnings, address deprecations

### PR DESCRIPTION
**Pull request by AetherUnbound**
## Description

This PR removes two filtered warnings for the Airflow/catalog tests that are no longer present, and addresses a deprecation warning that is currently being raised.

> [!NOTE]
> This targets #3572 and should be merged after it.

## Testing Instructions

1. `just catalog/test` shouldn't throw any warnings when the tests complete!

## Checklist

- [x] My pull request has a descriptive title (not a vague title like`Update index.md`).
- [x] My pull request targets the _default_ branch of the repository (`main`) or a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [x] I added or updated tests for the changes I made (if applicable).
- [x] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible errors.
- [ ] I ran the DAG documentation generator (if applicable).

[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

**Review by obulat — APPROVED** [review]
The CI is passing, and no warnings are present 👍

File: catalog/pytest.ini [hunk 1]
```diff
@@ -18,20 +18,13 @@ addopts =
...
-# airflow
-#   Warning within Airflow for an internal call, nothing we can do
 # flask
 #   https://docs.sqlalchemy.org/en/20/errors.html#error-b8d9
 #   Warning in dependency, nothing we can do
 #   "removed"/"remoevd" is due to https://github.com/pallets/flask
 # distutils
 #   Warning in dependency, nothing we can do
 filterwarnings=
-    ignore::sqlalchemy.exc.MovedIn20Warning
```
Comment (stacimc) [inline catalog/pytest.ini:33]: After rebasing locally and running `just catalog/test`, I see this MovedIn20Warning. Do we still need this filtered?

Comment (AetherUnbound) [reply catalog/pytest.ini:33]: Just to be sure, did you run a `just build` after rebasing?

Comment (stacimc) [reply catalog/pytest.ini:33]: I did a full `recreate`, actually, and confirmed locally that I was on Airflow v2.8.0 :/

Comment (AetherUnbound) [reply catalog/pytest.ini:33]: Oh my goodness, I realized I had `SQLALCHEMY_SILENCE_UBER_WARNING=1` in my catalog environment file 🤦🏼‍♀️ It seems reasonable to silence that in general so I'll add that to this PR! Thanks for pointing that out! The GHA test outputs also show that warning FWIW.

Comment (AetherUnbound) [reply catalog/pytest.ini:33]: Confirmed, warnings should now no longer be shown! Though you'll probably need to add that variable to your env file first Staci 😄

Comment (stacimc) [reply catalog/pytest.ini:33]: Oh, that'll do it! 😄 Looks great, makes sense to me.

**Review by stacimc — APPROVED** [review]

**Pull request merged**